### PR TITLE
SPCTR-2946 Fix: Forms - Not working when used within template part of FSE.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-2946

### Screenshots

- https://share.getcloudapp.com/7KuW1kWA

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Added to the card.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by RajkiranBagal — APPROVED** [review]

File: blocks-config/forms/class-uagb-forms.php [hunk 1]
```diff
@@ -72,6 +72,27 @@ public function reusable_block_content_on_page( $reusable_ref_id ) {
...
+				return;
+			}
+			$slug            = $block_attr['slug'];
+			$templates_parts = get_block_templates( array( 'slugs__in' => $slug ), 'wp_template_part' );
+			foreach ( $templates_parts as $templates_part ) {
+				if ( $slug === $templates_part->slug ) {
+					$id = $templates_part->wp_id;
+					return $id;
+				}
+			}
```
Comment (rahulvarma722) [inline blocks-config/forms/class-uagb-forms.php:93]:
```suggestion
$id = false;
			foreach ( $templates_parts as $templates_part ) {
				if ( $slug === $templates_part->slug ) {
					$id = $templates_part->wp_id;
					break;
				}
			}
			return $id;
```

File: blocks-config/forms/class-uagb-forms.php [hunk 2]
```diff
@@ -72,6 +72,27 @@ public function reusable_block_content_on_page( $reusable_ref_id ) {
...
+		/**
+		 * Generates ids for all wp template part.
+		 *
+		 * @param array $block_attr attributes array.
+		 * @since x.x.x
+		 * @return int
+		 */
+		public function get_fse_template_part( $block_attr ) {
+			if ( empty( $block_attr['slug'] ) ) {
+				return;
```
Comment (rahulvarma722) [inline blocks-config/forms/class-uagb-forms.php:84]:
```suggestion
				return false;
```

File: blocks-config/forms/class-uagb-forms.php [hunk 3]
```diff
@@ -72,6 +72,27 @@ public function reusable_block_content_on_page( $reusable_ref_id ) {
 			}
 		}
 
+		/**
+		 * Generates ids for all wp template part.
+		 *
+		 * @param array $block_attr attributes array.
+		 * @since x.x.x
+		 * @return int
```
Comment (rahulvarma722) [inline blocks-config/forms/class-uagb-forms.php:80]:
```suggestion
		 * @return integer|boolean
```

Comment (AbhijitNage123) [reply blocks-config/forms/class-uagb-forms.php]: Resolved

File: blocks-config/forms/class-uagb-forms.php (hunk 2 above)
Comment (AbhijitNage123) [reply blocks-config/forms/class-uagb-forms.php]: Resolved

File: blocks-config/forms/class-uagb-forms.php (hunk 1 above)
Comment (AbhijitNage123) [reply blocks-config/forms/class-uagb-forms.php]: Resolved

File: blocks-config/forms/class-uagb-forms.php [hunk 4]
```diff
@@ -72,6 +72,29 @@ public function reusable_block_content_on_page( $reusable_ref_id ) {
...
 
+		/**
+		 * Generates ids for all wp template part.
+		 *
+		 * @param array $block_attr attributes array.
+		 * @since x.x.x
+		 * @return integer|boolean
+		 */
+		public function get_fse_template_part( $block_attr ) {
+			$id = false;
```
Comment (rahulvarma722) [inline blocks-config/forms/class-uagb-forms.php:83]: Create $id after line number 86 and keep gap after line 86

**Review by rahulvarma722 — APPROVED** [review]

**Pull request merged**